### DISCUSSION TITLE
Wirable with v-model in HandlesActions

### DISCUSSION THREAD

**Discussion by satoved** [livewire/livewire#3974] [7. Bugs (archived)] [answered]
I was playing with the new Wireables.

Thought it would be great to typehint Enums.

I'm trying to use https://github.com/spatie/enum for the property:

_Enum class:_
```
class EstateTypeEnum extends Enum implements Wireable
{
    public function toLivewire()
    {
        return $this->value;
    }

    public static function fromLivewire($value)
    {
        return self::from($value);
    }
}
```

_Livewire component:_
```
class Test extends Component
{
    public EstateTypeEnum $e;

    public function mount()
    {
        $this->e = EstateTypeEnum::residence();
    }

    public function render()
    {
        return view('livewire.test');
    }
}
```

_test.blade.php:_
```
<div>
    <div>
        {{ $e->value }}
    </div>
    <select wire:model="e">
        @foreach(\App\Enums\EstateTypeEnum::toArray() as $value => $label)
            <option value="{{ $value }}">{{ $label }}</option>
        @endforeach
    </select>
</div>
```

**It works as expected:**
![image](https://user-images.githubusercontent.com/14319239/136567437-d1f2bf5a-2fe1-494a-b2b3-87c57da4c7f1.png)

When I change the value for the select, I expect it to be converted to my Enum class, and $this->e should become a new Enum class.

But I'm getting an error:
![image](https://user-images.githubusercontent.com/14319239/136567798-8f5128b6-22dc-4636-a8b8-9e8dfedd63d5.png)

As far as I can understand, it's trying to set a string value right from the select option value.

But isn't it supposed to check for _Wireable_ implementation and create an object out of it first inside _HandlesActions_?

**Comment by joshhanley** [comment, marked as answer]
@satoved currently the way the wireable DTO's are setup, is so that you can bind to properties on the DTO, not the whole DTO itself.

We're currently working on a solution for that, but for now, you can work around it by using an `updating` hook in your component and using value by reference `&$value`.

```php
public function updatingE(&$value) // Notice the pass by reference here
{
    $value = EstateTypeEnum::fromLivewire($value);
}
```

Hope this helps!

**Reply by satoved** [reply]
Thanks!

**Reply by joshhanley** [reply]
No worries! 🙂
